DoublyLinkedList/doubly_linked_list.h

```diff
@@ -0,0 +1,287 @@
+                        // to check whether the location is present or not
+                        //traversal till the condition is satisfied
+                    temp1 = temp1->next;
+                    count++; 
+            }
+                //new node has to be inserted after the temp1 node then temp1 is the previous node
+                //temp2 node will be next node
+            temp2 = temp1->next;
+                //as the new node is inserted between two temp nodes,
+            node* new_node=create_node(temp1, value, temp2);
+                //assigning values for the two before and after node.
+            temp1->next = new_node; 
+            temp2->prev = new_node;
+            length++;
+            return;
+        }
+    }
+}
+
+//for removing a node at front in the list
+int doubly_linked_list::remove_front(){
+    //to check whether the list is empty or not
+    if (head == nullptr){
+        cout << "No items in the list to delete"<<endl;
+        return 0;
+    }
+    //if the list has only one item
+    else if(head==tail){
+        int value = head->data;
+        delete (head);
+        head = tail = nullptr;
+        return value;
+    }
+    //if the list has two or more item
+    else{
+            // creating a temporary variable to store head node
+        node *temp = head;
+            //assigning the data to value in order to return
+        int value = temp->data;
+            //assigning the second nodes's prev pointer to null and change change head pointing to it
+        temp->next->prev = nullptr;
+        head = temp->next;
+            //clearing the node and decreasing the size
+        delete (temp);
+        length--;
+            //returning the value which is deleted 
+        return value;
+    }
+}
+//for removing last item in the list
+int doubly_linked_list::remove_back(){
+    if(tail==nullptr){
+        cout << "No items in the list to delete"<<endl;
+        return 0;
+    }
+    else if(head==tail){
+        int value;
+        delete (head);
+        head = tail = nullptr;
+        return value;
+    }
+    //if the list has two or more item
+    else{
+            // creating a temporary variable to store tail node
+        node *temp = tail;
+            //assigning the data to value in order to return
+        int value = temp->data;
+            //assigning the last second nodes's next pointer to null and change tail pointing to it
+        temp->prev->next = nullptr;
+        tail = temp->prev;
+            //clearing the node and decreasing the size
+        delete (temp);
+        length--;
+            //returning the value which is deleted 
+        return value;
+    }
+}
+//for removing node at a position
+void doubly_linked_list::remove(int location){
+    if(location>=length || location <0)
+    {
+        cout << "Invalid Location Specified to Remove"<<endl;
+    }
+    else
+    {
+        if (location == 0){
+            remove_front();
```

Comment:
instead of calling remove_front and remove_back from remove remove_front and remove_back should call remove(0) and remove(length-1) and the impl should happen here.